src/main/java/org/apache/sysds/runtime/instructions/cp/QuantileSortCPInstruction.java

```diff
@@ -119,4 +123,23 @@ public void processInstruction(ExecutionContext ec) {
 		//set and release output
 		ec.setMatrixOutput(output.getName(), resultBlock);
 	}
+
+	public int getNumThreads() {
+		return _numThreads;
+	}
+
+	@Override
+	protected FEDInstruction tryReplaceWithFederated(ExecutionContext ec) {
+		if(!input1.isMatrix())
+			return null;
+
+		MatrixObject mo1 = ec.getMatrixObject(input1);
+		if(!mo1.isFederatedExcept(FType.BROADCAST))
+			return null;
+
+		if(mo1.isFederated(FType.ROW) ||
+			(mo1.getFedMapping().getFederatedRanges().length == 1 && mo1.isFederated(FType.COL)))
+			return new QuantileSortFEDInstruction(this);
+		return null;
```

Comment:
As you can notice in all implementations I opted to use early return checks to reduce nesting. The last if on the other hand checks if replacement is appropriate. This is inconsistent to the checks before, which checked for failure, but is much more readable and therefore maintainable than confusing inverted conditions in my experience.